### ISSUE TITLE
ACF (Advanced Custom Fields) Checkboxes Field with WPAS Checkboxes Field

### ISSUE THREAD

**Issue by SolidDev** [bootsz/wp-advanced-search#137]
So I've noticed that when querying a meta_key of an ACF created field that is a select field, WPAS has no problem querying via either a select field or checkboxes. But when the ACF created meta_value is a checkbox field, WPAS won't successfully return any results when structured as a checkbox or a select field.

I am writing the 'checkbox fields' array like this:
     $args['fields'][] = array(  'type' => 'meta_key',
         'label' => 'Application',
         'format' => 'checkbox',
         'meta_key' => 'application',
         'compare' => 'IN',
         'values' => array('check 1' => 'Check 1', 'check 2' => 'Check 2', 'check 3' => 'Check 3',) 
     );

**Comment by SolidDev** [comment]
For clarity: 
ACF (Select) => WPAS (Select) SUCCESS
ACF (Select) => WPAS (Checkbox) SUCCESS
ACF (Checkbox) => WPAS (Select) FAIL
ACF (Checkbox) => WPAS (Checkbox) FAIL

**Comment by SolidDev** [comment]
I've also read your Plugins page that says when querying ACF fields that are arrays (checkboxes, multiselect), use the following in your 'meta_key' field's configuration:
    'data_type' => 'ARRAY<CHAR>',
    'compare' => 'LIKE',

So now I am trying to my field config looks like this:
    $args['fields'][] = array(  'type' => 'meta_key',
        'label' => 'Application',
        'format' => 'checkbox',
        'meta_key' => 'application',
        'data_type' => 'ARRAY<CHAR>',
         'compare' => 'LIKE',
        'values' => array('check 1' => 'Check 1', 'check 2' => 'Check 2', 'check 3' => 'Check 3'),
    );

Still no luck tho.

**Comment by bootsz** [comment]
it's 
```
data_type => 'ARRAY<CHAR>'
```

And you need to be using the version of the code from the 1.4-beta branch: https://github.com/bootsz/wp-advanced-search/tree/1.4-ajax

You can set ```$args['debug'] = true``` to turn on debugging and see what query arguments are being generated.

**Comment by SolidDev** [comment]
Hmm... So I've swapped out the 1.3 master files with the 1.4-ajax branch files and added $args['debug'] = true;, but now my form doesn't load. I also noticed that the ACF portion of the plugins page was just removed. Was that because of this thread?

**Comment by SolidDev** [comment]
I don't see any mention of a different way to construct the $args for 1.4-ajax. Am I missing something obvious?

**Comment by SolidDev** [comment]
I noticed that script.js wasn't loading properly, so I modified this line in init.php to fix that.
wp_enqueue_script( 'wpas-scripts', get_template_directory_uri() . '/library/' . basename(__DIR__) . '/js/scripts.js', array('jquery'), '1', false );
I've placed the wp-advanced-search directory in themes/mytheme/library/. The js loads without error now. Still no debugging msg tho.

**Comment by bootsz** [comment]
Alright my fault, I accidentally pasted the wrong branch.  This is the right one: https://github.com/bootsz/wp-advanced-search/tree/1.4-beta

I've deleted the other to avoid further confusion.  And yes, the way that forms are configured has changed slightly in the new version.  Details are here: http://wpadvancedsearch.com/beta/docs/setup/

Unless you're using AJAX you shouldn't need to do anything else other than set debug to true.  If using AJAX see http://wpadvancedsearch.com/beta/docs/setup/template-setup-ajax/

Since this is an upcoming version the docs are not 100% complete but they are close to done.  

I'll also look into seeing if we can make the directory path configurable for cases like yours where you want it in a sub-folder of your theme.

**Comment by SolidDev** [comment]
Success, great work! I'm up and running on the 1.4 beta. The AJAX sounds awesome, I'll have a look at that soon. Thanks!

**Comment by bootsz** [comment]
Great, glad it worked out
